### PR TITLE
DOC: clarify docstrings of `query_ball_tree` and `query_pairs`

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### What does this implement/fix?
I found some minor doc issues of `spatial.KDTree` and `spatial.cKDTree`.

I think current docstrings of `query_ball_tree` and `query_pairs` of `spatial.KDTree` and `spatial.cKDTree` are not good at below points:
1. docstrings are not consistent between `spatial.KDTree` and `spatial.cKDTree`.
2. `query_ball_tree` returns indexes of data, but docstring says that it find pairs of data.

So, I fixed docstrings of `query_ball_tree` and `query_pairs` of `spatial.KDTree` and `spatial.cKDTree`.

### CODE REVIEW

**Review by peterbell10 — CHANGES_REQUESTED** [review]

File: scipy/spatial/ckdtree.pyx [hunk 1]
```diff
@@ -995,7 +995,7 @@ cdef class cKDTree:
         """
         query_ball_tree(self, other, r, p=2., eps=0)
 
-        Find all pairs of points whose distance is at most r
+        Find all indexes of points whose distance is at most r
```
[the remark this answers is not in the document]
Comment (AtsushiSakai) [reply scipy/spatial/ckdtree.pyx]: Thank you for your comment. I got it. I updated docsting based on your comment. PTAL.

File: scipy/spatial/ckdtree.pyx [hunk 2]
```diff
@@ -995,7 +995,7 @@ cdef class cKDTree:
         """
         query_ball_tree(self, other, r, p=2., eps=0)
 
-        Find all pairs of points whose distance is at most r
+        Find all pairs of points between `self` tree and `other` tree whose distance is at most r
```
[the remark this answers is not in the document]
Comment (AtsushiSakai) [reply scipy/spatial/ckdtree.pyx]: OK. I fixed it. PTAL.

**Review by peterbell10 — APPROVED** [review]
LGTM, seems like a useful distinction.

**Comment by AtsushiSakai** [comment]
Thank you. If you have time, could you please review this PR also #12341 ? I think these functions should have Examples to understand its functionalities easily.

**Comment by AtsushiSakai** [comment]
[TODO] After merging this PR and #12341, we should answer this question:
python - Scipy cKDTree query_pairs versus query_ball_tree - Stack Overflow https://stackoverflow.com/questions/47763534/scipy-ckdtree-query-pairs-versus-query-ball-tree

**Comment by peterbell10** [comment]
Thanks @AtsushiSakai

**Pull request merged**